Source/Dafny/Compilers/Compiler-python.cs

```diff
@@ -989,15 +995,17 @@ protected class ClassWriter : IClassWriter {
     }
 
     protected override void EmitEmptyTupleList(string tupleTypeArgs, ConcreteSyntaxTree wr) {
-      throw new UnsupportedFeatureException(Token.NoToken, Feature.NonSequentializableForallStatements);
+      wr.Write("[]");
     }
 
     protected override ConcreteSyntaxTree EmitAddTupleToList(string ingredients, string tupleTypeArgs, ConcreteSyntaxTree wr) {
-      throw new UnsupportedFeatureException(Token.NoToken, Feature.NonSequentializableForallStatements);
+      var wrTuple = new ConcreteSyntaxTree();
+      wr.FormatLine($"{ingredients}.append(({wrTuple}))");
```

Comment:
How does this part work?  It looks like you're serializing `wrTuple` to string before it's been filled in?

Reply:
Yes, but I'm doing `FormatLine`, which will be evaluated when serialising the `ConcreteSyntaxTree`.

Reply:
TIL that `$"…"` is lazy sometimes.